app/main.py

```diff
@@ -4,18 +4,14 @@
 
 def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
     masks_to_buy = 0
-    try:
-        for friend in friends:
-            try:
-                cafe.visit_cafe(friend)
-            except NotWearingMaskError:
-                masks_to_buy += 1
-            except VaccineError:
-                raise
-    except VaccineError as e:
-        message = ''.join(list(e.args))
-        return message
-    else:
-        if masks_to_buy != 0:
-            return f"Friends should buy {masks_to_buy} masks"
-        return "Friends can go to KFC"
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotWearingMaskError:
+            masks_to_buy += 1
+        except VaccineError as e:
+            return str(e)
+
+    if masks_to_buy != 0:
+        return f"Friends should buy {masks_to_buy} masks"
+    return "Friends can go to KFC"
```

Comment:
Also, here function should return "Friends can go to {cafe.name}", not hard-coded KFC.

Reply:
Agree with you. Thanks for your vigilance!